### PR TITLE
Tweak: Making the animations CSS library to be loaded conditionally only when needed [ED-2749]

### PR DESCRIPTION
**Pull request by rotemee**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/base/elements-iteration-actions/assets.php [hunk 1]
```diff
@@ -138,6 +141,30 @@ private function is_active_page_assets_mode() {
...
+			$document_db_settings = [];
+		}
+
+		$document_current_settings = $this->document->get_settings();
+
+		if ( ! $document_current_settings ) {
+			$document_current_settings = [];
+		}
+
+		$document_settings = array_intersect_key( $document_db_settings, $document_current_settings );
```
Comment (kobizz) [inline core/base/elements-iteration-actions/assets.php:157]: 1. Use `array_replace_recursive`
2. Do it only on save

File: core/page-assets/loader.php [hunk 2]
```diff
@@ -24,7 +24,13 @@ public function get_name() {
 
 	private function init_assets() {
 		$this->assets = [
-			'styles' => [],
+			'styles' => [
+				'e-animations' => [
```
Comment (kobizz) [inline core/page-assets/loader.php:28]: Remove `wp_register_style` of animations

File: includes/frontend.php [hunk 3]
```diff
@@ -652,8 +652,9 @@ public function enqueue_styles() {
 			// The e-icons are needed in preview mode for the editor icons (plus-icon for new section, folder-icon for the templates library etc.).
 			if ( ! $this->is_improved_assets_loading() || Plugin::$instance->preview->is_preview_mode() ) {
 				wp_enqueue_style( 'elementor-icons' );
+				wp_enqueue_style( 'elementor-animations' );
```
Comment (kobizz) [inline includes/frontend.php:655]: Remove

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/page-assets/loader.php (hunk 2 above)
Comment (rotemee) [reply core/page-assets/loader.php:28]: Done.

**Pull request merged**